gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/services/reflect/DatabaseClasses.java

```diff
@@ -314,6 +314,23 @@ public final Class loadApplicationClass(ObjectStreamClass classDescriptor)
 		return loadApplicationClass(classDescriptor.getName());
 	}
 
+	public final Class loadClassFromDB(String name) throws ClassNotFoundException {
+		ClassNotFoundException cnfe = new ClassNotFoundException(name);
+		if (applicationLoader == null) throw cnfe;
+		// if DDL replay has not started then ignore
+		final GemFireStore memStore = GemFireStore
+				.getBootingInstance();
+		if (memStore == null || !(memStore
+				.initialDDLReplayInProgress()
+				|| memStore.initialDDLReplayDone())) {
+			throw cnfe;
+		}
+		Class c = applicationLoader.loadClass(name, true);
+		if (c == null)
+			throw cnfe;
+		return c;
+
+	}
 	public boolean isApplicationClass(Class theClass) {
```

Comment:
1. we do not need to check "com.pivotal.gemfirexd.internal" as the class loader is loading the classes first from the parent loader and if the current class loader is not able to load the class then only  it is loaded via this method.

2. cant use the existing method as it uses the Class.forName  and  which invokes the current ContextLoader and  ends up in infinite loop. having the check for #1 will also lead to the same issue.